### PR TITLE
fix: Profile Search with Search all fields has Undefined id

### LINKED ISSUE #2124: Profile Search with Search all fields has Undefined id
**Describe the bug**
Adding Search all fields on the profile search encounter undefined property

**To Reproduce**
Goto Buddyboss > Profiles > Profile Search Tab add (Search all fields ) fields.
[https://prnt.sc/yu4ae6](https://prnt.sc/yu4ae6)

Try to search/filter on members directory using search all fields textbox

**Expected behavior**
Notice: Undefined property: stdClass::$id in /Applications/XAMPP/xamppfiles/htdocs/buddyboss/wp-content/plugins/buddyboss-platform/bp-xprofile/bp-xprofile-repeaters.php on line 901

Notice: Undefined property: stdClass::$id in /Applications/XAMPP/xamppfiles/htdocs/buddyboss/wp-content/plugins/buddyboss-platform/bp-core/profile-search/bps-search.php on line 344

**Screenshots**
[https://prnt.sc/yu4lbl](https://prnt.sc/yu4lbl)
[https://prnt.sc/yu4ae6](https://prnt.sc/yu4ae6)

**Support ticket links**
https://secure.helpscout.net/conversation/1410111620/122532

### PR DESCRIPTION
**Pull request by milanhirapra**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #2124 .

### How to test the changes in this Pull Request:

1. Go to Buddyboss > Profiles > Profile Search Tab add (Search all fields ) fields. https://prnt.sc/yu4ae6
2. Try to search/filter on members directory using search all fields textbox

### Proof Screenshots or Video
https://vimeo.com/591067740/278137bc27

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

**Review by dipskakadiya — DISMISSED** [review]
@milanhirapra Approved. Ready for testing.

**Comment by JetalDobariya** [comment]
Hello @milanhirapra , @dipskakadiya 

I tested this PR in qa-ongoing branch, now "All search field" is working perfectly in the member directory.
See in the Video : https://www.loom.com/share/d67036c20bf9472481cddfccdb26fbf1

Thanks!

File: src/bp-xprofile/bp-xprofile-repeaters.php [hunk 1]
```diff
@@ -913,6 +913,10 @@ function bp_view_profile_repeaters_print_group_html_end() {
  * @param object $f Passed by reference
  */
 function bp_profile_repeaters_search_change_filter( $f ) {
+	if( ! isset( $f->id ) ){
```
Comment (chetansatasiya) [inline src/bp-xprofile/bp-xprofile-repeaters.php:916]:
```suggestion
	if ( ! isset( $f->id ) ) {
```

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
@milanhirapra Follow the spacing properly.

**Review by KartikSuthar — APPROVED** [review]

**Comment by JetalDobariya** [comment]
Hello @milanhirapra and @dipskakadiya 

I did verify this PR in qa-approved branch, now "Search with All" field results are getting perfect on Member Directory page.
--> https://www.loom.com/i/cfc77bcb5fff4eae858ac960d17b33d7

Thanks!

**Pull request merged**